src/krb5_tgs_fmt_plug.c

```diff
@@ -122,15 +122,37 @@ static int valid(char *ciphertext, struct fmt_main *self)
 			p = strtokm(ctcopy, "*");
 			ctcopy += strlen(p) + 2;	/* set after '$' */
 			goto edata;
-		}	
+		}
 		if (ctcopy[0] == '$')
 			ctcopy++;
 	}
 
 edata:
+
 	if (((p = strtokm(ctcopy, "$")) == NULL) || strlen(p) != 32)	/* assume checksum */
 		goto err;
-		/* assume edata2 following */
+	if (!ishex(p))
+		goto err;
+	/* assume edata2 following */
+	p += 33;
+	if (!ishex(p) || strlen(p) < 500) {
+		/* handle 'chopped' .pot lines */
+		/* since this format has an inconsistant signature, we do not
+		 * use the common ldr_isa_pot_source, sense we can not validate
+		 * that this hash is not some other format from the .pot file
+		 * that has 32 byte hex, and $ and is a chopped pot line.
+		 */
+		int len;
+		if (!ldr_in_pot || strlen(ciphertext) != LINE_BUFFER_SIZE)
+			goto err;
+		len = hexlen(p);
+		if (len < LINE_BUFFER_SIZE - 100 || len > LINE_BUFFER_SIZE)
+			goto err;
+		p += len;
+		if (*p != '$' || !strstr(p, LDR_TRIMMED_POT_BIN_SIG))
+			goto err;
+		return 1;
+	}
```

Comment:
There's some hard-coded arbitrary (?) figures here we should probably avoid (or at least add a FIXME comment for later review) or document. How were "500" and "100" chosen?